### PR TITLE
feat: writing squid chapter

### PR DESCRIPTION
**Pull request by alemorvan**
#### Author checklist (Completed by original Author)
- [x] Good fit for the Rocky Linux project? Title and Author Metatags inserted ?
- [x] If applicable, steps and instructions have been tested to work
- [x] Initial self-review to fix basic typos and grammar completed

#### Rocky Documentation checklist (Completed by Rocky team) 
- [x] 1st Pass (Document is good fit for project and author checklist completed)
- [x] 2nd Pass (Technical Review - check for technical correctness) 
- [x] 3rd Pass (Detailed Editorial Review and Peer Review)
- [x] Final approval (Final Review)

### CODE REVIEW

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+
+The client workstation is protected by two firewalls and never communicates directly with the outside network.
+
+![Proxy-based architecture](img/squid-architecture-proxy.png)
+
+!!! NOTE
+
+    This architecture requires browser configuration on the client workstation.
+
+With a captive proxy, there's no need to configure all client workstations.
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:48]:
```suggestion
With a captive proxy, there is no need to configure all client workstations.
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+    This architecture requires a specific configuration on the router.
+
+In the case of standard proxy or captive proxy architecture, one of the primary interests of this type of service is, of course, to act as a cache.
+
+In this way, a file downloaded once from the WAN (and therefore potentially from a slower link than the LAN) is stored in memory by the proxy-cache, to be used by subsequent clients. In this way, the bandwidth of the slow link is optimized.
+
+As we'll see later in this chapter, this isn't the only use for a proxy.
+
+A proxy can be deployed to :
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:64]:
```suggestion
A proxy can be deployed to:
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+| 2XX  | Success                |
+| 3XX  | Redirection            |
+| 4XX  | Customer request error |
+| 5XX  | Server error           |
+
+Examples:
+
+* 200 : ok
+* 301 : Moved Permanently
+* 302 : Moved Temporatly
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:103]:
```suggestion
* 302 : Moved Temporarily
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+* 304 : Not modified
+* 400 : Bad request
+* 401 : Unauthorized
+* 404 : Not found
+
+#### About Squid
+
+Squid supports http and ftp protocols.
+
+The advantages of installing a solution based on the Squid server :
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:113]:
```suggestion
The advantages of installing a solution based on the Squid server:
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+
+#### Squid server tree and files
+
+The single configuration file is `/etc/squid/squid.conf`.
+
+Service logs (stop and restart) are stored in `/var/log/squid.cache.log`, while client requests are stored in `/var/log/squid/access.log`. By default, cache files are stored in `/var/spool/squid/`.
+
+#### The `squid` command
+
+The `squid` command control the squid server.
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:149]:
```suggestion
The `squid` command controls the squid server.
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 6]
```diff
@@ -0,0 +1,458 @@
...
+```
+
+| Option      | Description                  |
+|-------------|------------------------------|
+| `-z`        | Initialize cache directories |
+| `-s`        | Enables syslog logging       |
+| `-k parse`  | Test configuration file      |
+| `-k rotate` | Rotates logs                 |
+
+Logging client requests can quickly lead to the storage of large volumes of data.
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:164]:
```suggestion
Logging client requests can quickly lead to the storage of large amount of data.
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 7]
```diff
@@ -0,0 +1,458 @@
...
+
+When the service is restarted, the Squid server will listen on the port defined by the http_port directive.
+
+* RAM reservation `cache_mem`
+
+```bash
+cache_mem taille KB|taille MB|taille GB
+```
+
+For example :
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:192]:
```suggestion
For example:
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 8]
```diff
@@ -0,0 +1,458 @@
...
+```
+
+| Option | Description              |
+|--------|--------------------------|
+| ufs    | Unix File System         |
+| 100    | Size in mega             |
+| 16     | 16 top-level folders     |
+| 256    | 256 second-level folders |
+
+When the service is launched for the first time, the :
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:293]: ???

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 9]
```diff
@@ -0,0 +1,458 @@
...
+
+#### Les Access Control List (ACL)
+
+Syntax of the `http_access` directive
+
+```bash
+http_access allow|deny [!]acl_name
+```
+
+Example :
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:310]:
```suggestion
Example:
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 10]
```diff
@@ -0,0 +1,458 @@
...
+
+Syntax of the `acl` directive:
+
+```bash
+acl name type argument
+```
+
+The order of ACLs is cumulative. Several ACLs with the same name represent a single ACL.
+
+Examples :
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:327]:
```suggestion
Examples:
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md [hunk 11]
```diff
@@ -0,0 +1,458 @@
...
+| Option | Description                                   |
+|--------|-----------------------------------------------|
+| `-s`   | Silent mode (displays nothing in the console) |
+| `-h`   | Define target proxy                           |
+| `-p`   | Listening port (default 3128)                 |
+| `-r`   | Force server to reload object                 |
+
+#### Analyze logs
+
+Squid's log records can be monitored with the command :
```
Comment (SergeCroise) [inline docs/books/web_services/053-load-balancer-proxies-squid.md:414]:
```suggestion
Squid's log records can be monitored with the command:
```

File: docs/books/web_services/053-load-balancer-proxies-squid.md (hunk 1 above)
Comment (SergeCroise) [reply docs/books/web_services/053-load-balancer-proxies-squid.md:48]: see the section **Avoid Contractions** in the documentation:
https://docs.rockylinux.org/guides/contribute/style_guide/#voice-and-tone

**Review by sspencerwire — APPROVED** [review]

**Pull request merged**